### ISSUE TITLE
java.lang.AbstractMethodError: Method io/flutter/project/FlutterProjectSystem.getSourceProvidersFactory()Lcom/android/tools/idea/projectsystem/SourceProvidersFactory; is abstract from Android Studio 4.0 Canary 4

### ISSUE THREAD

**Issue by devoncarew** [flutter/flutter-intellij#4226]
We're seeing lots of these internally:

Flutter plugin version `42.1.2`

`AndroidStudio` / `4.0 Canary 4`

```
java.lang.AbstractMethodError: Method io/flutter/project/FlutterProjectSystem.getSourceProvidersFactory()Lcom/android/tools/idea/projectsystem/SourceProvidersFactory; is abstract
	io.flutter.project.FlutterProjectSystem.getSourceProvidersFactory(), FlutterProjectSystem.java:-1
	org.jetbrains.android.facet.SourceProviderManagerKt.createSourceProviderFor(), SourceProviderManager.kt:114
	org.jetbrains.android.facet.SourceProviderManagerKt.getSourceProviderManager(), SourceProviderManager.kt:99
	org.jetbrains.android.facet.SourceProviderManager$Companion.getInstance(), SourceProviderManager.kt:45
	org.jetbrains.android.facet.SourceProviderManager.getInstance(), SourceProviderManager.kt:-1
	org.jetbrains.android.facet.AndroidRootUtil.getPrimaryManifestFile(), AndroidRootUtil.java:79
	com.android.tools.idea.model.MergedManifestContributors$Companion.determineFor(), MergedManifestContributors.kt:59
	com.android.tools.idea.model.MergedManifestContributors.determineFor(), MergedManifestContributors.kt:-1
	com.android.tools.idea.model.MergedManifestInfo.create(), MergedManifestInfo.java:185
	com.android.tools.idea.model.MergedManifestSupplier$getOrCreateSnapshot$$inlined$runReadAction$1.compute(), actions.kt:78
	com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(), ApplicationImpl.java:918
	com.android.tools.idea.model.MergedManifestSupplier.getOrCreateSnapshot(), MergedManifestManager.kt:349
	com.android.tools.idea.model.MergedManifestSupplier.getOrCreateSnapshotInCallingThread(), MergedManifestManager.kt:142
	com.android.tools.idea.model.MergedManifestManager$Companion.getFreshSnapshot(), MergedManifestManager.kt:334
	com.android.tools.idea.model.MergedManifestManager$Companion.getSnapshot(), MergedManifestManager.kt:313
	com.android.tools.idea.model.MergedManifestManager$Companion.getSnapshot(), MergedManifestManager.kt:300
	com.android.tools.idea.model.MergedManifestManager.getSnapshot(), MergedManifestManager.kt:-1
	com.android.tools.idea.model.AndroidModuleInfo.getPackage(), AndroidModuleInfo.java:106
	com.android.tools.idea.ddms.DevicePanel.getApplicationName(), DevicePanel.java:205
	com.android.tools.idea.ddms.DevicePanel.access$000(), DevicePanel.java:49
	com.android.tools.idea.ddms.DevicePanel$1.compute(), DevicePanel.java:66
	com.android.tools.idea.ddms.DevicePanel$1.compute(), DevicePanel.java:62
	com.intellij.openapi.util.NullableLazyValue.getValue(), NullableLazyValue.java:36
	com.android.tools.idea.ddms.DevicePanel.getPreferredClient(), DevicePanel.java:395
	com.android.tools.idea.ddms.DevicePanel.updateProcessComboBox(), DevicePanel.java:348
	com.intellij.openapi.application.TransactionGuardImpl$2.run(), TransactionGuardImpl.java:312
	com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(), LaterInvocator.java:433
	com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(), LaterInvocator.java:416
	com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(), LaterInvocator.java:399
	java.awt.event.InvocationEvent.dispatch(), InvocationEvent.java:311
	java.awt.EventQueue.dispatchEventImpl(), EventQueue.java:764
	java.awt.EventQueue.access$500(), EventQueue.java:98
	java.awt.EventQueue$3.run(), EventQueue.java:715
	java.awt.EventQueue$3.run(), EventQueue.java:709
	java.security.AccessController.doPrivileged(), AccessController.java:-2
	java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(), ProtectionDomain.java:74
	java.awt.EventQueue.dispatchEvent(), EventQueue.java:734
	com.intellij.ide.IdeEventQueue.defaultDispatchEvent(), IdeEventQueue.java:824
	com.intellij.ide.IdeEventQueue._dispatchEvent(), IdeEventQueue.java:773
	com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(), IdeEventQueue.java:412
	com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(), CoreProgressManager.java:704
	com.intellij.ide.IdeEventQueue.dispatchEvent(), IdeEventQueue.java:411
	java.awt.EventDispatchThread.pumpOneEventForFilters(), EventDispatchThread.java:205
	java.awt.EventDispatchThread.pumpEventsForFilter(), EventDispatchThread.java:116
	java.awt.EventDispatchThread.pumpEventsForHierarchy(), EventDispatchThread.java:105
	java.awt.EventDispatchThread.pumpEvents(), EventDispatchThread.java:101
	java.awt.EventDispatchThread.pumpEvents(), EventDispatchThread.java:93
	java.awt.EventDispatchThread.run(), EventDispatchThread.java:82
```

**Comment by stevemessick** [comment]
The 42.1.2 version indicates this plugin was built for AS 3.6 beta. It would be nice if we could find a way to prevent beta plugins from being installed into canary IDEs. But when both beta and canary Android Studio are built from the same 192.x base it is difficult. I haven't found a way yet. Even with the dev channel this will still be a potential problem.

**Comment by stevemessick** [comment]
Technically, this was fixed in #4267. The underlying issue I described before still exists, so I don't know if you want to close this or not.
